src/pull.go

```diff
@@ -82,6 +92,76 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type renameOp struct {
+		newName      string
+		change       *Change
+		originalPath string
+	}
+	clashesMap := map[string][]*Change{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil {
+			continue
+		}
+		clashesMap[clash.Path] = append(group, clash)
+	}
+
+	renames := make([]renameOp, 0, len(clashesMap)) // we will have at least len(clashesMap) renames
```

Comment:
It is a premature optimization that could introduce bugs, if we chose at all to not return on encountering the error and instead reComposeError, all of a sudden we have nil renames and have to modify the source after a bug hunt :( also I'd rather make the optimization only after benchmarking and noticing a performance hit for most purposes we won't notice a difference.